soc/arm/Kconfig

```diff
@@ -1,4 +1,8 @@
 # Copyright (c) 2024 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
+if SOC_FAMILY_ARM || SOC_FAMILY_ARM64
```

Comment:
this exactly shows why safe guarding at this level can fail in some cases and why guarding should be done in the specific Kconfigs.

Because if family is of type `arm64`, then `arm` Kconfigs are also sourced (which they should not be).
Thus a safeguard at this level gives false safety.

In this case, the following setting is technically defined as an int but as the default is properly having an `if`, then no harm.
Technically the setting is defined when building for `fvp_base_revc_2xaemv8a`:
https://github.com/zephyrproject-rtos/zephyr/blob/9076ac64e95cba7fd013dee212f731c4430f1573/soc/arm/mps3/Kconfig#L18-L20